molgenis-fair/src/main/java/org/molgenis/fair/controller/EntityModelWriter.java

```diff
@@ -0,0 +1,158 @@
+import org.eclipse.rdf4j.model.impl.LinkedHashModel;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.molgenis.data.Entity;
+import org.molgenis.data.meta.model.Attribute;
+import org.molgenis.data.meta.model.EntityType;
+import org.molgenis.data.semantic.LabeledResource;
+import org.molgenis.data.semantic.Relation;
+import org.molgenis.data.semanticsearch.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.xml.datatype.DatatypeConfigurationException;
+
+import static com.google.common.collect.Iterables.contains;
+import static java.util.Arrays.stream;
+
+@Component
+public class EntityModelWriter
+{
+	private static final String KEYWORD = "http://www.w3.org/ns/dcat#keyword";
+
+	private final SimpleValueFactory valueFactory;
+	private final TagService<LabeledResource, LabeledResource> tagService;
+
+	@Autowired
+	public EntityModelWriter(TagService<LabeledResource, LabeledResource> tagService, SimpleValueFactory valueFactory)
+			throws DatatypeConfigurationException
+	{
+		this.valueFactory = valueFactory;
+		this.tagService = tagService;
+	}
+
+	private void setNamespacePrefixes(Model model)
+	{
+		model.setNamespace("rdf", "http://www.w3.org/1999/02/22-rdf-syntax-ns#");
+		model.setNamespace("rdfs", "http://www.w3.org/2000/01/rdf-schema#");
+		model.setNamespace("dcat", "http://www.w3.org/ns/dcat#");
+		model.setNamespace("xsd", "http://www.w3.org/2001/XMLSchema#");
+		model.setNamespace("owl", "http://www.w3.org/2002/07/owl#");
+		model.setNamespace("dct", "http://purl.org/dc/terms/");
+		model.setNamespace("lang", "http://id.loc.gov/vocabulary/iso639-1/");
+		model.setNamespace("fdpo", "http://rdf.biosemantics.org/ontologies/fdp-o#");
+		model.setNamespace("ldp", "http://www.w3.org/ns/ldp#");
+		model.setNamespace("foaf", "http://xmlns.com/foaf/0.1/");
+		model.setNamespace("orcid", "http://orcid.org/");
+	}
+
+	public Model createRdfModel(String subjectIRI, Entity objectEntity)
+	{
+		Resource subject = valueFactory.createIRI(subjectIRI);
+		Model model = new LinkedHashModel();
+		setNamespacePrefixes(model);
+		EntityType entityType = objectEntity.getEntityType();
+
+		for (Attribute objectAttribute : entityType.getAtomicAttributes())
+		{
+			Object value = objectEntity.get(objectAttribute.getName());
+			if (value == null)
+			{
+				continue;
+			}
+			for (LabeledResource tag : tagService.getTagsForAttribute(entityType, objectAttribute)
+					.get(Relation.isAssociatedWith))
+			{
+				IRI predicate = valueFactory.createIRI(tag.getIri());
+				addRelationForAttribute(model, subject, predicate, objectEntity, objectAttribute);
+			}
+		}
+		return model;
+	}
+
+	private void addRelationForAttribute(Model model, Resource subject, IRI predicate, Entity objectEntity,
+			Attribute objectAttribute)
+	{
+		String name = objectAttribute.getName();
+
+		switch (objectAttribute.getDataType())
```

Comment:
Include all cases and throw a RuntimeException for the default case. This prevents missing case statement warnings and alerts us that we have to decide what to do when introducing new data types.